Small mention about disjunctive patterns

AHBruns opened the pull request:
Disjunctive patterns are super valuable when you (rightly) don't want to use a wildcard, but have a lot of cases that match to the same thing. I think they should at least be mentioned in the docs even if they may not warrant a whole sub-section.

Because of the tiny size of this change, I just went ahead and opened a PR since it took about the same amount of work as opening an issue on the topic. I hope that's okay.

AHBruns commented:
Ah jeez. I'm sorry about trigger so many builds!

IwanKaramazow commented on pages/docs/manual/latest/pattern-matching-destructuring.mdx:
````diff
@@ -580,6 +580,35 @@ Here are some advices.
+| E => 2
+}
+```
+This still requires a bit more work than a wildcard, but we promise it'll be worth it.
````
`we promise it'll be worth it.` -> this is super vague. Can you illustrate/explain this?

IwanKaramazow replied:
Also the rescript formatter allows the following kind of formatting:
```
switch v {
| A => 1
| B | C | D | E => 2
}
```

IwanKaramazow commented on pages/docs/manual/latest/pattern-matching-destructuring.mdx:
```diff
@@ -580,6 +580,35 @@ Here are some advices.
 
 Do not abuse the wildcard `_` too much. This prevents the compiler from giving you better exhaustiveness check, which would be especially important after a refactoring where you add a new case to a variant. Try only using `_` against infinite possibilities, e.g. string, int, etc.
 
+If you're about to use a wildcard when matching on a large but finite set of possibilities because you have many possibilities that return the same thing, consider reaching for disjunctive patterns instead. That is, you can replace:
```
This sentence is very long. Can you divide it into shorter sentences?
Is it worth introducing the term `disjunctive patterns`?  What if we just talk about "writing out all cases explicitly"?

AHBruns replied:
I'll rewrite it to be more approachable. My only reason for using the term was to give readers a specific term that they can do more research on.

This brings up the larger idea of a glossary section which gives uncommon terms and brief definitions allowing people to grok concepts as well as giving them specific keywords with which to do independent research. This probably deserves an issue and longer discussion though.

AHBruns commented:
I refactored the intro to be simpler, and only mentioned the term disjunctive patterns as a brief aside. I also cleaned up the code examples and added a note about the refactoring advantages (mainly that you can have multiple disjunctive pattern sets unlike wildcards).

This is getting a bit large. So, it might be reasonable to split it into its own sub-section, but I'll leave that decision to you, of course.

ryyppy commented:
I like it! Maybe it would make sense to put this content further up in the the "Exhaustiveness Check" Section with a sub header like "Match multiple cases to one expression" or similar? That way it would not get lost in the Tips / Tricks section

AHBruns commented:
Moved it to its own sub-section. LMK if you'd like any other tweaks.

chenglou commented:
Hey @AHBruns. Thanks for the effort; this is indeed valuable to document more explicitly. I'll go ahead and merge it, but I'll rewrite it a bit.

chenglou commented:
Done. See my comments at https://github.com/reason-association/rescript-lang.org/commit/db82e572f2942a7e6c0d011df9102277610e14e8.

Thanks again!
